pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/FunctionsImpl.java

```diff
@@ -686,10 +686,33 @@ public void updateFunctionOnWorkerLeader(final String tenant,
         }
 
         // Redirect if we are not the leader
+        boolean shouldRedirect = false;
         if (!worker().getLeaderService().isLeader()) {
             WorkerInfo workerInfo = worker().getMembershipManager().getLeader();
-            URI redirect = UriBuilder.fromUri(uri).host(workerInfo.getWorkerHostname()).port(workerInfo.getPort()).build();
-            throw new WebApplicationException(Response.temporaryRedirect(redirect).build());
+            // there might be a delay between when the leader service for this worker gets triggered "becomeActive"
```

Comment:
I think we should adopt more simple workflow. 
1. LeaderService.isLeader() is set only after we have fully become leader.
2. If we are leader, do the request ourselves
3. else re-direct to the leader
There will be a case that we might redirect to a leader who is not ready. In that case just return error to the user asking him to retry again is perfectly valid.

Reply:
I would rather minimize the number of errors we return to the user